File: Frontend/FrontendApp/src/app/models/dog.ts
export class Dog {
    id: number;
    name: string;
    breed: string;
    age: number;
    size: string;
    color: string;
    specialMark: string = "";
    hairLength: string;
    earsType: string;
    tailLength: string;
    behaviors: string[] = [''];

    constructor(id: number, name: string, breed: string, age: number, size: string, color: string, specialMark: string,
        hairLength: string, earsType: string, tailLength: string, behaviors: string[]) {
        this.id = id;
        this.name = name;
        this.breed = breed;
        this.age = age;
        this.size = size;
        this.color = color;
        this.specialMark = specialMark;
        this.hairLength = hairLength;
        this.earsType = earsType;
        this.tailLength = tailLength;
        this.behaviors = behaviors;
    }
}

File: Backend/Backend/Models/Dogs/Dog.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Backend.Models.Dogs
{
    public abstract class Dog
    {
        [Required]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Breed { get; set; }
        [Required]
        public int Age { get; set; }
        [Required]
        [MaxLength(50)]
        public string Size { get; set; }
        [Required]
        [MaxLength(100)]
        public string Color { get; set; }
        [Required]
        [MaxLength(100)]
        public string SpecialMark { get; set; }
        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [Required]
        public PictureDog Picture { get; set; }
        [Required]
        [MaxLength(50)]
        public string HairLength { get; set; }
        [Required]
        [MaxLength(50)]
        public string EarsType { get; set; }
        [Required]
        [MaxLength(50)]
        public string TailLength { get; set; }
        [Required]
        public List<DogBehavior> Behaviors { get; set; }
    }
}
